Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -67,6 +78,32 @@ Ref<FragmentedSharedBuffer> FragmentedSharedBuffer::create(DataSegment::Provider
     return adoptRef(*new FragmentedSharedBuffer(WTFMove(provider)));
 }
 
+#if PLATFORM(COCOA)
+std::optional<Ref<FragmentedSharedBuffer>> FragmentedSharedBuffer::fromIPCData(IPCData&& ipcData)
+{
+    return WTF::switchOn(WTFMove(ipcData), [](Vector<std::span<const uint8_t>>&& data) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        size_t size = sizeof(data[0]) * data.size();
```

Comment:
Accessing `data[0]` looks dangerous here, especially considering the data comes from IPC and is thus untrusted.
Als, sizeof(data[0]) is always 1 anyway since it's a uint8_t.

Reply:
Also, if you truly want to get the size of the data (and I think you do for proper validation), you'll need to iterate the vector and sum the size of each span it contains.